gulpfile.js

```diff
@@ -56,54 +56,57 @@ gulp.task('example-serve', ['browserify', 'styles', 'watch'], function() {
     });
 });
 
-gulp.task('tests-browserify', function() {
-    return browserify('./test/tests')
-        .bundle()
-        .on('error', function (err) {
-            console.log(err.toString());
-            this.emit('end');
-        })
-        .pipe(source('tests.js'))
-        .pipe(gulp.dest('test/build/'));
-});
-
-gulp.task('test', ['tests-browserify'], function () {
-    var browserTests = gulp
-    .src('./test/test-runner.html')
-    .pipe(mochaPhantomJS({
+function gulp_test() {
+    return gulp.src([
+        'test/**/*.spec.js'
+    ])
+    .pipe(mocha({
         log: true,
         timeout: 10000,
         slow: 3000,
         reporter: 'spec',
-        ui: 'bdd'
+        ui: 'bdd',
+        require: ['./test/init.js']
     }));
+}
 
-    var nodeTests = gulp
-    .src([
-        'test/lib/**/*.spec.js',
-
-        // exclude lib browser tests
-        '!test/lib/charts/*.spec.js',
-        '!test/lib/generators/*.spec.js',
-        '!test/lib/components/*.spec.js'
+gulp.task('instrument', function () {
+    return gulp.src([
+        'src/**/*.js',
+        '!src/lib/components/log-explorer/**/*.js'
```

Comment:
let me check as I think the ispart thing isn't necessary and I may have **copy & pasta'd** a little too much.